lib/logstash/filters/translate.rb

```diff
@@ -114,11 +124,15 @@ class LogStash::Filters::Translate < LogStash::Filters::Base
 
   public
   def register
+    registering = true
     if @dictionary_path
       @next_refresh = Time.now + @refresh_interval
-      registering = true
       load_yaml(registering)
     end
+    if @dictionary_url
```

Comment:
you are totally right, it must be exclusive, it have not sense make the two work on the same "instance". Thanks! changing!